include/boost/url/bnf/parse.hpp

```diff
@@ -77,214 +98,182 @@ parse(
-   #include <boost/url/error.hpp>
-
-   namespace boost {
-   namespace urls {
-   namespace detail {
-   // BNF element to parse a non-empty sequence of alpha chars ([a-zA-Z]+)
-   struct alpha_string_bnf {
-     // reference to a std::string where we should store the results
-     std::string &v;
-
-     // parsing an alpha_string_bnf element
-     friend bool parse(char const *&it, char const *end, error_code &ec,
-                       alpha_string_bnf &&t) {
-       // is_alpha functor
-       bnf::alpha_chars is_alpha;
-
-       // check if empty
-       if (it == end) {
-         ec = error::incomplete;
-         return false;
-       }
-
-       // check if not alpha at all
-       if (!is_alpha(*it)) {
-         ec = error::bad_alpha;
-         return false;
-       }
-       t.v.push_back(*it);
-       ++it;
-
-       // consume more alphas
-       while (it != end && is_alpha(*it)) {
-         t.v.push_back(*it);
-         ++it;
-       }
-       ec = error::success;
-       return true;
-     }
-   };
-   } // namespace detail
-   } // namespace urls
-   } // namespace boost
-
-   int main() {
-     using namespace boost::urls;
-
-     // Input
-     std::string str = "Hello.cpp";
-     auto it = str.data();
-     auto const end = it + str.size();
-
-     // Output
-     std::string alpha_l;
-     std::string alpha_r;
-     error_code ec;
-
-     // Parse two alpha strings separated by a '.'
-     if (bnf::parse(it, end, ec, detail::alpha_string_bnf{alpha_l}, '.',
-                    detail::alpha_string_bnf{alpha_r})) {
-       std::cout << alpha_l << std::endl;
-       std::cout << alpha_r << std::endl;
-     }
-     return 0;
+   using bnf::parse;
+   if (bnf::parse(it, end, ec, bnf_element_1, bnf_element_2, bnf_element_3)) {
+     std::cout << "Range [it, end) parsed successfully\n";
    }
    @endcode
 
    @par Exception Safety
-     Calls to `parse` overloads may throw, although they should
-     prefer setting @ref ec instead.
-
-   @see https://en.wikipedia.org/wiki/Backus%E2%80%93Naur_form
 
-        @ref boost::urls::error
+     Defined by the types of `args`.
 
-   @return `true` if the range initial chars match the elements successfully,
-   `false` otherwise
+   @return `true` if the range initial chars match the elements successfully
 
-   @param[in,out] it An iterator to the first element in the range.
-   At the end of this function, @ref it points to one past the last
+   @param it An iterator to the first element in the range.
+   At the end of this function, `it` points to one past the last
    element parsed in the range.
 
-   @param[in] end An iterator to one past the last element in the range
+   @param end An iterator to one past the last element in the range
+
+   @param ec Set to the error, if any occurred
+
+   @param args BNF elements to be parsed
 
-   @param[out] ec The error code this function should set if it fails
+   @see
 
-   @param[out] args BNF elements to be parsed
+   @ref error
```

Comment:
indent and period

Reply:
See: https://github.com/CPPAlliance/url/blob/02ef2ae540a10466bc7462ac4534fad2d711c627/include/boost/url/url_view.hpp#L1911